nntrainer/layers/embedding.cpp

```diff
@@ -81,7 +87,8 @@ void EmbeddingLayer::forwarding(RunLayerContext &context, bool training) {
   TensorDim out_tensor_dim = TensorDim({1, 1, 1, out_dim});
 
   for (unsigned int b = 0; b < input_.batch(); ++b) {
-    float *in_data = input_.getAddress(b * input_.getDim().getFeatureLen());
+    float *in_data =
+      (float *)input_.getAddress(b * input_.getDim().getFeatureLen());
```

Comment:
same as above (suggest to use template T=float)

Reply:
Currently wip for supporting various datatype cases for layers.
fyi, works like
```C++
TensorDim out_tensor_dim = TensorDim({1, 1, 1, out_dim});
```
to 
```C++
...
TensorDim::TensorType tensor_type = {context.getFormat(), context.getWeightDataType()}
...
TensorDim out_tensor_dim = TensorDim({1, 1, 1, out_dim}, tensor_type );
...
```
is also working in process, for every layer